File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_20_state/java/_02_after/Student.java
package com.example._03_behavioral_patterns._20_state.java._02_after;

import java.util.HashSet;
import java.util.Set;

public class Student {

    private final String name;

    public Student(String name) {
        this.name = name;
    }

    private final Set<OnlineCourse> onlineCourses = new HashSet<>();

    public boolean isAvailable(OnlineCourse onlineCourse) {
        return onlineCourses.contains(onlineCourse);
    }

    public void addPrivate(OnlineCourse onlineCourse) {
        this.onlineCourses.add(onlineCourse);
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                '}';
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_20_state/kt/_02_after/Student.kt
package com.example._03_behavioral_patterns._20_state.kt._02_after

class Student(
    private val name: String,
    private val onlineCourses: MutableSet<OnlineCourse> = HashSet()
) {
    fun isAvailable(onlineCourse: OnlineCourse): Boolean {
        return onlineCourses.contains(onlineCourse)
    }

    fun addPrivate(onlineCourse: OnlineCourse) {
        onlineCourses.add(onlineCourse)
    }

    override fun toString(): String {
        return "Student{" +
                "name='" + name + '\'' +
                '}'
    }
}
